eclipse-sw360/sw360
# feat(rest): Update API create Release with Moderator, Contributor, CpeId

[//]: # (Copyright Bosch.IO GmbH 2020)
[//]: # (This program and the accompanying materials are made)
[//]: # (available under the terms of the Eclipse Public License 2.0)
[//]: # (which is available at https://www.eclipse.org/legal/epl-2.0/)
[//]: # (SPDX-License-Identifier: EPL-2.0)

> Please provide a summary of your changes here.
> * Which issue is this pull request belonging to and how is it solving it? (*Refer to issue here*)
> * Did you add or update any new dependencies that are required for your change?


### How To Test?

- Run command line: 
```
curl 'https://sw360.org/api/releases' -i -X POST \
    -H 'Content-Type: application/hal+json' \
    -H 'Authorization: Bearer eyJhbGciOiJSUzI1NiIsInR5cCI6IkpXVCJ9.eyJhdWQiOlsic3czNjAtUkVTVC1BUEkiXSwidXNlcl9uYW1lIjoiYWRtaW5Ac3czNjAub3JnIiwic2NvcGUiOlsiYWxsIl0sImV4cCI6MTY5MjAwNDYyOSwiYXV0aG9yaXRpZXMiOlsiUkVBRCIsIldSSVRFIl0sImp0aSI6IkxoS1l1aEItbUJSeVV1QWNLT2FseXRSWW1yYyIsImNsaWVudF9pZCI6InRydXN0ZWQtc3czNjAtY2xpZW50In0.pObj6HM4dKGeEK0weAX8TLGt3Da9nrkXxFwbjdLsEylnRID1gDax_NonMYvHAI8gEqRzZtE7FJ2CAhveJiSIH8rwzDh8jzbJL-o0G3NRZylmYBtv4uAe8rbSkNrLuzRrjgtorQyl7AlUdwlAd4VoCVAeJ5iB1-kIgrrMrsFk-6K3yOtxSCF3IlMIbq8NuR-UthkiwM1DuQaBIF-Zcf5Dv8rJU-MQkAuKeqrrXxpo_SHsJqdBa4t59oTH5g_GtFEvKJw2Ww3dTzbmBf844JSmVcNx6eH7CV1DdhL6DcGXcMk_d4Bfz0vPKaHHunoIUvBgZ76RZM7SCYyP5q5PS2hrYw' \
    -d '{
  "componentId": "17653524",
  "version": "1.0",
  "contributors": [
      "user@sw360.org",
      "clearing_admin1@sw360.org"
  ],
  "moderators": [
      "clearing_admin1@sw360.org",
      "admin@sw360.org"
  ],
  "cpeid": "cpe:2.3:a:s:glibc:1.0.0"
}'

```

- Expected Output

```
{
    "name": "Test Release",
    "version": "1.0",
    "cpeid": "cpe:2.3:a:s:glibc:1.0.0",
    "contributors": [
        "user@sw360.org",
        "clearing_admin1@sw360.org"
    ],
    "_links": {
        "sw360:component": {
            "href": "https://sw360.org/api/components/17653524"
        },
        "self": {
            "href": "https://sw360.org/api/releases/1234567890"
        },
        "curies": [
            {
                "href": "https://sw360.org/docs/{rel}.html",
                "name": "sw360",
                "templated": true
            }
        ]
    }
	"_embedded": {
        "sw360:moderators": [
            {
                "email": "clearing_admin1@sw360.org",
                "deactivated": false,
                "fullName": "Test Clearing1",
                "_links": {
                    "self": {
                        "href": "http://localhost:8080/resource/api/users/byid/cecb575b594a495e8df56067368f70e7"
                    }
                }
            },
            {
                "email": "admin@sw360.org",
                "deactivated": false,
                "fullName": "Test Admin",
                "_links": {
                    "self": {
                        "href": "http://localhost:8080/resource/api/users/byid/3309473e580848e8b32dc2bdd2ddf1ba"
                    }
                }
            }
        ]
    }
}
```
